README.md

```diff
@@ -1,124 +1,106 @@
 
-#### Convert `Optional<T>` to `T?`
-
-```kotlin
-// If Optional is None — you'll get null, otherwise you'll get not null T value.
+// Optional<T> → T?
+// If optional is None — you'll get null, otherwise you'll get non-null T value.
 val t = optional.toNullable()
 ```
 
-#### Get `value` or fall back to something else if `None` (`getOrElse()`)
+### Leverage Kotlin Features
+
+#### Fallback from `None` (like `java.util.Optional.getOrElse()`)
 
 ```kotlin
 val f = optional.toNullable() ?: "fallback"
 ```
-#### Check if `Optional` is `Some` or `None`
+#### Use [Smart Cast](http://kotlinlang.org/docs/reference/typecasts.html#smart-casts)
 
 ```kotlin
 when (optional) {
-    is Some -> {}
-    is None -> {}
+    is Some -> println(optional.value)
+    is None -> println("Nope!")
 }
 ```
 
-#### Destructure `Optional<T>`
-
-Koptional supports [destructuring](https://kotlinlang.org/docs/reference/multi-declarations.html).
-
-Destructuring has same effect as calling `toNullable()`.
+#### Use [Destructuring](https://kotlinlang.org/docs/reference/multi-declarations.html)
 
 ```kotlin
-val o: Optional<T> = something.toOptional()
-
-// If Optional is None — you'll get null, otherwise you'll get not null T value.
-val (value) = o
+// If Optional is None — you'll get null, otherwise you'll get non-null T value.
+val (value) = optional
 ```
 
-#### Filter only `Some` values emitted by RxJava 2 or Project Reactor
+### Interop with Java
 
-```kotlin
-val values: Observable<String> = Observable
-    .just(Some("a"), None, Some("b"))
-    .filterSome()
-    .map { value: String -> } // filterSome() unwraps value for you.
-```
+Use the static `Optional.toOptional()` to wrap an instance of `T` into `Optional<T>`.
 
-#### Filter only `None` values emitted by RxJava 2 or Project Reactor
+### Work with RxJava 2
 
 ```kotlin
-val noneSignals: Observable<Unit> = Observable
-    .just(Some("a"), None, Some("b"))
-    .filterNone()
-    .map { none: Unit -> } // filterNone() maps None to Unit.
-```
+val values = Observable.just(Some("a"), None, Some("b"))
 
-### Interoperability with Java
+// Filter Some values.
+values
+    .filterSome()
+    .test()
+    .assertValues("a", "b")
 
-Use the static `Optional.toOptional()` method (declared as a companion object method) to wrap an 
-instance of `T` into `Optional<T>`.  
+// Filter None values.
+values
+    .filterNone()
+    .test()
+    .assertValues(Unit) // filterNone() maps None to Unit.
+```
 
-### Download
+## Download
 
 Koptional is [available on jcenter](https://jcenter.bintray.com/com/gojuno/koptional).
 
-Optional type:
+`Optional`:
 
 ```groovy
-compile 'com.gojuno.koptional:koptional:put-some-version'
+implementation "com.gojuno.koptional:koptional:$koptional_version"
 ```
 
 [RxJava 2][rxjava2] extensions:
 
 ```groovy
-compile 'com.gojuno.koptional:koptional-rxjava2-extensions:put-some-version'
+implementation "com.gojuno.koptional:koptional-rxjava2-extensions:$koptional_version"
 ```
 
```

Comment:
Can you add Reactor pls?

Reply:
Sure. Nuked a separate `README.md` and inlined it here as well.